src/markup/codemirror/create.ts

```diff
@@ -157,7 +161,49 @@ export function createCodemirror(params: CreateCodemirrorParams) {
                 onScroll(event);
             },
             paste(event, editor) {
-                if (event.clipboardData && parseInsertedUrlAsImage) {
+                if (!event.clipboardData) return;
+
+                // if clipboard contains YFM content - avoid any meddling with pasted content
+                // since text/yfm will contain valid markdown
+                const yfmContent = event.clipboardData.getData(DataTransferType.Yfm);
+                if (yfmContent) {
+                    event.preventDefault();
+                    editor.dispatch(editor.state.replaceSelection(yfmContent));
+                    return;
+                }
+
+                // checking if a copy buffer content is suitable for convertion
+                const shouldSkipHtmlConversion = shouldSkipHtmlConversionBasedOnClipboard(
+                    event.clipboardData,
+                );
+
+                // if we have text/html inside copy/paste buffer
+                const htmlContent = event.clipboardData.getData(DataTransferType.Html);
+                // if we pasting markdown from VsCode we need skip html transformation
+                if (htmlContent && !disableHTMLParsingInMd && !shouldSkipHtmlConversion) {
```

Comment:
And why is this variable needed? To maintain the old state of the editor? or there are cases when it is necessary

I also suggest calling it `parseHtmlOnPaste` and making it `false` by default (for backward compability)

I suggest moving it from the experiments to the [markup config](https://github.com/gravity-ui/markdown-editor/blob/ed85f7116e419451bb7bad19ac4c96d88d0ee7cc/src/bundle/types.ts#L166)

Reply:
I need it to skip major update in package (since this can be a breaking change for some edge cases)